### PR TITLE
feat: Provide Configuration Options for POB

### PR DESCRIPTION
**Pull request by keefertaylor**
When integrating into Duality we discovered that not all chains will have a Distribution Module.

To account for this, add a new abstraction, `RewardsAddressProvider` that will instruct the moduled as to where auction rewards should go. There are two options provided:
1. `ProposerRewardsAddressProvider`: Send rewards to the block proposer
2. `FixedAddressRewardsAddressProvider`: Send rewards to the same address

This PR uses a second function to construct the POB module for backwards compatibility.

Notes:
- The default dependency injection strategy for the module will use the `Proposer` provider, chains can override this preference
- There is boilerplate code to inject a `Fixed` provider. This will default to the module's address and chains can provide their own injection binding if they want something else. 
- I didn't see a unit test for auction profits, so I haven't added an additional test.

### CODE REVIEW

File: api/pob/builder/v1/genesis.pulsar.go [hunk 1]
```diff
@@ -455,7 +455,6 @@ var (
 	fd_Params_max_bundle_size          protoreflect.FieldDescriptor
 	fd_Params_escrow_account_address   protoreflect.FieldDescriptor
 	fd_Params_reserve_fee              protoreflect.FieldDescriptor
-	fd_Params_min_buy_in_fee           protoreflect.FieldDescriptor
```
Comment (bpiv400) [inline api/pob/builder/v1/genesis.pulsar.go:458]: Why are we removing this? 

I think it _should_ be gone but I'm surprised it's happening in this PR

Comment (bpiv400) [reply api/pob/builder/v1/genesis.pulsar.go:458]: It looks like there's not a diff in the corresponding protos, so it seems like @davidterpay @nivasan1 just didn't regen these files as part of cutting the previous release. 

This should probably happen as a part of CI

File: x/builder/keeper/keeper.go [hunk 2]
```diff
@@ -8,22 +8,23 @@ import (
 	"github.com/cosmos/cosmos-sdk/codec"
 	storetypes "github.com/cosmos/cosmos-sdk/store/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	rewardsaddressprovider "github.com/skip-mev/pob/x/builder/rewards_address_provider"
```
Comment (bpiv400) [inline x/builder/keeper/keeper.go:11]: incredibly irrelevant nit: It seems common to abbreviate when importing long module names.

**Review by bpiv400 — APPROVED** [review]
One thing that seems missing: The ability to query for the rewards for address

Should we add this to the querier?

File: api/pob/builder/v1/genesis.pulsar.go (hunk 1 above)
Comment (keefertaylor) [reply api/pob/builder/v1/genesis.pulsar.go:458]: Filed https://linear.app/skip/project/pob-65f1cb76c755/ENG

**Comment by keefertaylor** [comment]
Filed https://linear.app/skip/issue/ENG-1382/add-a-query-for-the-rewards-address to query the rewards address

**Comment by davidterpay** [comment]
have these changes also been merged into main?

**Pull request merged**